app/main.py

```diff
@@ -1 +1,15 @@
-# write your code here
+class CommandError(Exception):
+    pass
+
+
+def copy_file(command: str) -> None:
+    command_list = command.split(" ")
+
+    if command_list[0] == "cp":
+        with (
+            open(command_list[1], "r") as file_in,
+            open(command_list[2], "w") as file_out,
```

Comment:
currently, if file names are the same, line 11 will immediately overwrite the file, and your reader variable later will be an empty string. Which will be then written to the file